src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java

```diff
@@ -116,6 +120,28 @@
     return promise.future().map(updateResult -> updateResult.rowCount() == 1);
   }
 
+  @Override
+  public Future<List<JobExecution>> getExpiredEntries(Date expirationDate, String tenantId) {
+    Promise<Results<JobExecution>> promise = Promise.promise();
+    try {
+      Criterion expiredEntriesCriterion = constructExpiredEntriesCriterion(expirationDate);
+      pgClientFactory.getInstance(tenantId).get(TABLE, JobExecution.class, expiredEntriesCriterion, false, promise);
+    } catch (Exception e) {
+      LOGGER.error("Error during getting fileDefinition entries by expired date", e);
+      promise.fail(e);
+    }
+    return promise.future().map(Results::getResults);
 
+  }
+
+  private Criterion constructExpiredEntriesCriterion(Date expirationDate) {
+    Criterion criterion = new Criterion();
+    Criteria lastUpdateDateCriteria = new Criteria();
```

Comment:
Shouldn't we check if the job status is in progress. What if the job already completed?

Reply:
sure, will add